Fc 1146 event sources (#7)
Templum/developers-community: pages/Documents/DeveloperTools/LivePersonFunctions/EventSources/chat-post-survey.md

indicator: both
---

> Chat Post Survey integrates LivePerson Functions with LiveEngage Chat. This integration will use Functions to re-create one of our Legacy features, the ability to conditionally send out email transcripts after a chat conversation ended.

Reviewer: Is chat or messaging the legacy

Author: Is Chat

{: .important}
Within this integration, the chat server will still take care of sending out emails. Functions will only be used to write more sophisticated conditions.

### Configuration

> There is not further requirements than FaaS feature enabled. Remember that this integration is available only with LiveEngage Chat.

Reviewer: ```suggestion
There is no further requirements than that  FaaS was [provisioned](link to provisioning). Remember that this integration is available only with LiveEngage Chat.
```

Reviewer: Doube-check if this is still the real name. Given we have conversation cloud now and not Live engange

Reviewer: Still relevant

#### Step 1 - Create a new Function

> Create a new `lambda` from **Chat Post Survey E-mail Transcript** event. The default template for this event will be selected for the function.

Reviewer: Here you can link to deep dive

{: .important}
Once a Function is implemented for this specific event, any old logic outside of Functions will not be used anymore (i.e, any rules defined otherwise in Conversational Cloud). Therefore, make sure that the complete logic was implemented within your function before deployment.

#### Step 2 - Edit the Function

Adjust the coding from the template according to your needs by modifying the lambda/function by pressing the edit button on the template function. The email result entry needs to contain the following properties:

<table>
  <thead><tr><th>Property</th><th>description</th><th>type</th></tr></thead>
  <tbody>
    <tr><td>targetEmail</td><td>Receiving email address.</td><td>STRING</td></tr>
    <tr><td>senderName</td><td>The name of the sender of the post-survey email transcript.</td><td>STRING</td></tr>
    <tr><td>senderEmail</td><td>Sender email address.</td><td>STRING</td></tr>
    <tr><td>subject</td><td>Email subject.</td><td>STRING</td></tr>
  </tbody>
</table>

#### Step 3 - Deploy the Function

> Just like any other function, this function must be deployed before it can be used. [Please see this document](liveperson-functions-getting-started-your-first-function.html#deploy) for more information on how to deploy your function.

Reviewer: Link to deep dive

### Function example

> Based on the payload sent from the Conversational Cloud server described bellow, this could be an example `lambda` that processes the data within LivePerson Functions and returns a list of email addresses back to Conversational Cloud:

Reviewer: There is no payload shown. So don't refer to it

Reviewer: So after seeing it. You should restructure the whole paragraph. First show the Payload (without the need to have a subheader).

Then you introduce your example function with the text (while changing the wording) and finally you show the code.

```javascript
module.exports = (data, cb) => {
  let allSurveyEmails = [];

  //csatRank value range: 1-5, 1=very dissatisfied, 5=very satisified
  let csatRank = data.payload.csatRank > -1 ? data.payload.csatRank : null;
  let skillName = data.payload.skillName;
  let agentName = data.payload.agentName;
  let questionsAndAnswers = data.payload.questionWithAnswers;

  if(csatRank && csatRank <= 3 && data.payload.skillName.match(/CustomerCare/i) !== null && data.payload.agentName.match(/John Doe/i) !== null){
    let surveyEmail = {};
    surveyEmail.targetEmail = "ReplaceMe@company.com";
    surveyEmail.senderName = "Sender X";
    surveyEmail.senderEmail = "ReplaceMe@company.com";
    surveyEmail.subject = "Email Transcript Notification";
    allSurveyEmails.push(surveyEmail);
  }

  //return result
  cb(null, allSurveyEmails);

> };

Reviewer: This is not even a legit lambda in terms of our current function. Please re-write it

Reviewer: Also module.exports is not encouraged by us. So avoid referring to it

Author: fixed

<tr><td>agentName</td><td></td><td>Full name of the agent in the format "first name surname".</td><td>STRING</td><td>John Doe</td></tr>
<tr><td>questionWithAnswers</td><td></td><td>List of all survey questions which were answered.</td><td>STRING</td><td></td></tr>
<tr><td>questionWithAnswers</td><td>Question</td><td>Full question string.</td><td>STRING</td><td>How was the chat experience?</td></tr>
<tr><td>questionWithAnswers</td><td>Answer</td><td>Complete answer. For multi-selection, the answers will be separated by a semicolon.</td><td>STRING</td><td>Poor;Below Average</td></tr>
</tbody></table>

The following is an example of the payload that Conversational Cloud Chat sends to Functions in a JSON format:

```json
{
  "timestamp": 2342343242,
  "headers": [],
  "payload": {
    "csatRank": 3,
    "skillName": "CustomerCare",
    "agentName": "John Doe",
    "questionWithAnswers": [
    {
      "question": "How was the chat experience?",
      "answer": "Poor;Below Average"
    }
    ]
  }
}
```

### Hints

* Only **Chat Post Survey E-mail Transcript** event is supported for this integration.

> * Functions platform won't send out any email. The chat server will take care of sending out emails.

Reviewer: ```suggestion
* Functions platform won't send out any email. Instead the triggering service will send out the email based on the function response.
```

* Any old logic/rules related to this event outside the Functions will be replaced by the function once it is deployed. Be sure to implement the completed logic to avoid unexpected behaviors.

> * If there is any error during an invocation, the email won't be sent, there is no retry mechanism in the chat server for this integration.

Reviewer: ```suggestion
* If there is any error during an invocation, the email won't be sent, there is **no retry** mechanism in the chat server for this integration.
```
